packages/miniflare/src/plugins/hyperdrive/index.ts

```diff
@@ -90,8 +90,24 @@ export const HYPERDRIVE_PLUGIN: Plugin<typeof HyperdriveInputOptionsSchema> = {
 			}
 		);
 	},
-	getNodeBindings() {
-		return {};
+	getNodeBindings(options) {
+		return Object.fromEntries(
+			Object.entries(options.hyperdrives ?? {}).map(([name, url]) => {
+				const connectionOverrides: Record<string | symbol, string> = {
+					connectionString: `${url}`,
+					port: url.port,
+					host: url.host,
+				};
+				const proxyNodeBinding = new ProxyNodeBinding({
+					get(target, prop) {
+						return prop in connectionOverrides
+							? connectionOverrides[prop]
+							: target[prop];
+					},
+				});
+				return [name, proxyNodeBinding];
+			})
+		);
```

Comment:
However, I don't really understand why we need this new `ProxyNodeBinding()` class. What about just:

```ts
getNodeBindings(options) {
		return Object.fromEntries(
			Object.entries(options.hyperdrives ?? {}).map(([name, url]) => {
				return [name, {
					connectionString: url.href,
					port: url.port,
					host: url.host,
				}];
			})
		);
```

Reply:
The point of `ProxyNodeBinding` is that I wanted to add a new way that can generally be applied to all bindings to allow the overriding of some property accessors.

In this specific case we use the class to override `connectionString`, `port` and `host` but do allow everything else to pass through, which, here is just `connect` method (which I haven't tested yet to be honest).

I think this could be quite nice as it will enable us to simply override any proxy behavior in the future as (/if) the need arises.